### PR TITLE
Add science GCSE to support interface

### PR DESCRIPTION
**Pull request by tomas-stefano**
## Context

Add science GCSE edit to support interface.

## Changes proposed in this pull request

Similar to #8741 

## Guidance to review

1. Does it work?

## Link to Trello card

https://trello.com/c/rnhWVd0w/837-support-edit-gcses-functionality

### CODE REVIEW

File: app/forms/concerns/science_gcse_helper.rb [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+  end
+
+  def science?
+    subject == ApplicationQualification::SCIENCE
+  end
+
+  def gsce_qualification_type?
+    qualification.qualification_type == 'gcse'
+  end
+  alias gcse? gsce_qualification_type?
```
Comment (vassyz) [inline app/forms/concerns/science_gcse_helper.rb:156]: Why do you need this alias?

Comment (tomas-stefano) [reply app/forms/concerns/science_gcse_helper.rb:156]: gsce_qualification_type? is used in candidate interface and I prefer to use gcse? in support (that's the name of the option)

Comment (vassyz) [reply app/forms/concerns/science_gcse_helper.rb:156]: Ok, then you should be able to remove [this method](https://github.com/DFE-Digital/apply-for-teacher-training/pull/8749/files#diff-6911382f09dcdbb528cdd81044ce846fc528ea0c2d5add24427ee32669630d40R185). And you got a typo `gsce_` -> `gcse_`

File: spec/system/support_interface/change_science_gcse_spec.rb [hunk 2]
```diff
@@ -0,0 +1,489 @@
...
+      institution_country: nil,
+      other_uk_qualification_type: nil,
+      non_uk_qualification_type: nil,
+      enic_reference: nil,
+      comparable_uk_qualification: nil,
+    )
+  end
+
+  def then_it_should_save_uk_o_level_into_the_application_form
+    expect(page).to have_content('GCSE updated')
```
Comment (vassyz) [inline spec/system/support_interface/change_science_gcse_spec.rb:320]: THoughts on putting
```ruby
expect(page).to have_content('GCSE updated')

@science_gcse.reload
expect(@science_gcse.attributes.symbolize_keys).to include({...})
```
inside a method?

Comment (tomas-stefano) [reply spec/system/support_interface/change_science_gcse_spec.rb]: Done!

Comment (tomas-stefano) [reply spec/system/support_interface/change_science_gcse_spec.rb]: Done

File: spec/system/support_interface/change_science_gcse_spec.rb [hunk 3]
```diff
@@ -0,0 +1,489 @@
...
+  end
+
+  def and_i_click_to_change_my_science_gcse
+    within('.app-edit-qualification') do
+      click_link 'Change'
+    end
+  end
+  alias_method :when_i_click_to_change_my_science_gcse, :and_i_click_to_change_my_science_gcse
+
+  def when_i_click_update
```
[the remark this answers is not in the document]
Comment (tomas-stefano) [reply spec/system/support_interface/change_science_gcse_spec.rb]: Done

File: spec/system/support_interface/change_science_gcse_spec.rb [hunk 4]
```diff
@@ -0,0 +1,489 @@
...
+  def then_i_see_a_change_science_link
+    expect(page).to have_link('Change')
+  end
+
+  def and_i_click_to_change_my_science_gcse
+    within('.app-edit-qualification') do
+      click_link 'Change'
+    end
+  end
+  alias_method :when_i_click_to_change_my_science_gcse, :and_i_click_to_change_my_science_gcse
```
[the remark this answers is not in the document]
Comment (tomas-stefano) [reply spec/system/support_interface/change_science_gcse_spec.rb]: Done

File: app/forms/concerns/science_gcse_helper.rb (hunk 1 above)
Comment (tomas-stefano) [reply app/forms/concerns/science_gcse_helper.rb:156]: Done! The problem with the typo is that the typo is on candidate interface and I am hesitant to change candidate interface at this point tbh.

**Review by vassyz — APPROVED** [review]

**Pull request merged**